### ISSUE TITLE
poetry shell/run automatically installs current pyproject.toml

### ISSUE THREAD

**Issue by blackliner** [python-poetry/poetry#5134]
Working in a monorepo leads to frequent changes, and currently one has to remember to rerun `poetry install` when changing branches. I did not find a way to let `poetry shell` or `poetry run` do a check if its venv is up2date with the lockfile, and perform an install in case it is not. That would be a nice convenience feature.

- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

**Comment by timoguin** [comment]
Agreed. Having a way to explicitly trigger install when doing `poetry shell` or `poetry run` would be quite helpful to me right now.
